src/main/java/teamrtg/rtg/modules/bop/biomes/RTGBiomeBOPCherryBlossomGrove.java

```diff
@@ -0,0 +1,105 @@
+import biomesoplenty.api.biome.BOPBiomes;
+import biomesoplenty.common.block.BlockBOPLeaves;
+import biomesoplenty.common.block.BlockBOPLog;
+import biomesoplenty.common.enums.BOPTrees;
+import biomesoplenty.common.enums.BOPWoods;
+import net.minecraft.init.Biomes;
+import net.minecraft.init.Blocks;
+import teamrtg.rtg.api.tools.deco.DecoBoulder;
+import teamrtg.rtg.api.tools.deco.DecoFallenTree;
+import teamrtg.rtg.api.tools.deco.helper.DecoHelperRandomSplit;
+import teamrtg.rtg.api.world.RTGWorld;
+import teamrtg.rtg.api.world.biome.TerrainBase;
+import teamrtg.rtg.api.world.biome.deco.DecoBase;
+import teamrtg.rtg.api.world.biome.deco.DecoBaseBiomeDecorations;
+import teamrtg.rtg.api.world.biome.surface.part.CliffSelector;
+import teamrtg.rtg.api.world.biome.surface.part.SurfacePart;
+import teamrtg.rtg.modules.bop.RTGBiomeBOP;
+
+public class RTGBiomeBOPCherryBlossomGrove extends RTGBiomeBOP {
+
+    public RTGBiomeBOPCherryBlossomGrove() {
+        super(BOPBiomes.cherry_blossom_grove.get(), Biomes.RIVER);
+    }
+
+    @Override
+    public TerrainBase initTerrain() {
+        return new TerrainBase() {
+
+            private float minHeight = 58f;
+            private float maxHeight = 76f;
+            private float hillStrength = 20f;
+
+            {
+                this.minHeight = minHeight;
+                this.maxHeight = (maxHeight > rollingHillsMaxHeight) ? rollingHillsMaxHeight : ((maxHeight < this.minHeight) ? rollingHillsMaxHeight : maxHeight);
+                this.hillStrength = hillStrength;
+            }
+
+            @Override
+            public float generateNoise(RTGWorld rtgWorld, int x, int y, float biomeWeight, float border, float river) {
+                return terrainRollingHills(x, y, rtgWorld.simplex, river, hillStrength, maxHeight, groundNoise, groundNoiseAmplitudeHills, 0f);
+            }
+        };
+    }
+
+    @Override
+    public SurfacePart initSurface() {
+        SurfacePart surface = new SurfacePart();
+        surface.add(new CliffSelector(1.5f)
+            .add(PARTS.STONE_OR_COBBLE));
+        surface.add(PARTS.surfaceGeneric());
+        return surface;
+    }
+
+    @Override
+    public void initDecos() {
+
+        DecoBoulder decoBoulder = new DecoBoulder();
+        decoBoulder.boulderBlock = Blocks.COBBLESTONE.getDefaultState();
+        decoBoulder.chance = 16;
+        decoBoulder.maxY = 95;
+        this.addDeco(decoBoulder);
+
+        DecoFallenTree decoFallenTree1 = new DecoFallenTree();
+        decoFallenTree1.distribution.noiseDivisor = 100f;
+        decoFallenTree1.distribution.noiseFactor = 6f;
+        decoFallenTree1.distribution.noiseAddend = 0.8f;
+        decoFallenTree1.logCondition = DecoFallenTree.LogCondition.NOISE_GREATER_AND_RANDOM_CHANCE;
+        decoFallenTree1.logConditionNoise = 0f;
+        decoFallenTree1.logConditionChance = 16;
+        decoFallenTree1.maxY = 100;
+        decoFallenTree1.logBlock = BlockBOPLog.paging.getVariantState(BOPWoods.CHERRY);
```

Comment:
There's probably a better way to access BOP's blocks than this:

```
decoFallenTree1.logBlock = BlockBOPLog.paging.getVariantState(BOPWoods.CHERRY);
decoFallenTree1.leavesBlock = BlockBOPLeaves.paging.getVariantState(BOPTrees.PINK_CHERRY);
```

In 1.7.10, we accessed the block registry and grabbed the block by its name, but I wasn't sure how to do that with BOP's wood & leaves as they seemed to be setup differently than regular blocks?